packages/editor-ui/src/components/MainHeader/WorkflowDetails.vue

```diff
@@ -1,16 +1,535 @@
+import type { BaseTextKey } from '../../plugins/i18n';
+
+const props = defineProps<{
+	workflow: IWorkflowDb;
+	readOnly?: boolean;
+}>();
+
+const $style = useCssModule();
+
+const rootStore = useRootStore();
+const canvasStore = useCanvasStore();
+const settingsStore = useSettingsStore();
+const sourceControlStore = useSourceControlStore();
+const tagsStore = useTagsStore();
+const uiStore = useUIStore();
+const usersStore = useUsersStore();
+const workflowsStore = useWorkflowsStore();
+
+const router = useRouter();
+const route = useRoute();
+
+const locale = useI18n();
+const telemetry = useTelemetry();
+const message = useMessage();
+const toast = useToast();
+const titleChange = useTitleChange();
+const workflowHelpers = useWorkflowHelpers({ router });
+
+const isTagsEditEnabled = ref(false);
+const isNameEditEnabled = ref(false);
+const appliedTagIds = ref<string[]>([]);
+const tagsEditBus = createEventBus();
+const tagsSaving = ref(false);
+const importFileRef = ref<HTMLInputElement | undefined>();
+const eventBus = createEventBus();
+
+const hasChanged = (prev: string[], curr: string[]) => {
+	if (prev.length !== curr.length) {
+		return true;
+	}
+
+	const set = new Set(prev);
+	return curr.reduce((acc, val) => acc || !set.has(val), false);
+};
+
+const isNewWorkflow = computed(() => {
+	return (
+		!props.workflow.id ||
+		props.workflow.id === PLACEHOLDER_EMPTY_WORKFLOW_ID ||
+		props.workflow.id === 'new'
+	);
+});
+
+const isWorkflowSaving = computed(() => {
+	return uiStore.isActionActive('workflowSaving');
+});
+
+const onWorkflowPage = computed(() => {
+	return route.meta && (route.meta.nodeView || route.meta.keepWorkflowAlive === true);
+});
+
+const onExecutionsTab = computed(() => {
+	return [
+		VIEWS.EXECUTION_HOME.toString(),
+		VIEWS.WORKFLOW_EXECUTIONS.toString(),
+		VIEWS.EXECUTION_PREVIEW,
+	].includes((route.name as string) || '');
+});
+
+const workflowPermissions = computed(() => {
+	return getWorkflowPermissions(usersStore.currentUser, props.workflow);
+});
+
+const workflowMenuItems = computed<IActionDropdownItem[]>(() => {
+	const actions: IActionDropdownItem[] = [
+		{
+			id: WORKFLOW_MENU_ACTIONS.DOWNLOAD,
+			label: locale.baseText('menuActions.download'),
+			disabled: !onWorkflowPage.value,
+		},
+	];
+
+	if (!props.readOnly) {
+		actions.unshift({
+			id: WORKFLOW_MENU_ACTIONS.DUPLICATE,
+			label: locale.baseText('menuActions.duplicate'),
+			disabled: !onWorkflowPage.value || !props.workflow.id,
+		});
+
+		actions.push(
+			{
+				id: WORKFLOW_MENU_ACTIONS.IMPORT_FROM_URL,
+				label: locale.baseText('menuActions.importFromUrl'),
+				disabled: !onWorkflowPage.value || onExecutionsTab.value,
+			},
+			{
+				id: WORKFLOW_MENU_ACTIONS.IMPORT_FROM_FILE,
+				label: locale.baseText('menuActions.importFromFile'),
+				disabled: !onWorkflowPage.value || onExecutionsTab.value,
+			},
+		);
+	}
+
+	if (hasPermission(['rbac'], { rbac: { scope: 'sourceControl:push' } })) {
+		actions.push({
```

Comment:
Can we disable this also if environments are not enabled to avoid such error messages:
![image](https://github.com/n8n-io/n8n/assets/2598782/1eeb2c90-c126-428d-9d3d-32098e7815f1)

Reply:
Update: Discussed this, in my case Source control was enabled (and in turn the menu item) but environments weren't so the decision was to keep the error message for clarity and visibility